pkg/subnet/kube/kube.go

```diff
@@ -634,3 +637,53 @@ func (m *kubeSubnetManager) HandleSubnetFile(path string, config *subnet.Config,
 	}
 	return subnet.WriteSubnetFile(path, config, ipMasq, sn, ipv6sn, mtu)
 }
+
+// GetStoredMacAddress reads MAC address from node annotations when flannel restarts
+func (ksm *kubeSubnetManager) GetStoredMacAddress() string {
+	// The kube subnet mgr needs to know the k8s node name that it's running on so it can annotate it.
+	// If we're running as a pod then the POD_NAME and POD_NAMESPACE will be populated and can be used to find the node
+	// name. Otherwise, the environment variable NODE_NAME can be passed in.
+	nodeName := os.Getenv("NODE_NAME")
```

Comment:
`kubeSubnetManager struct` already includes the node (nodeName field), can't you use that one?

Reply:
you are absolutely right!